src/eba_handlers.eliom

```diff
@@ -203,22 +203,29 @@ let activation_handler_common ~restart ~akey =
     let%lwt userid = Eba_user.userid_of_activationkey akey in
     let%lwt () = Eba_db.User.set_email_validated userid in
     let%lwt () = Eba_session.connect userid in
-    Lwt.return ()
+    Lwt.return `Reload
   with Eba_db.No_such_resource ->
     Eliom_reference.Volatile.set
       Eba_userbox.activation_key_outdated true;
     Eba_msg.msg ~level:`Err ~onload:true
       "Invalid activation key, ask for a new one.";
-    Lwt.return ()
+    Lwt.return `NoReload
 
 let%server activation_handler akey () =
-  let%lwt () = activation_handler_common ~restart:false ~akey in
-  Eliom_registration.Action.send ()
+  let%lwt a = activation_handler_common ~restart:false ~akey in
```

Comment:
Note that `~restart` is never used by `activation_handler_common`... In a branch of mine, I've removed `~restart` from `activation_handler_common`.